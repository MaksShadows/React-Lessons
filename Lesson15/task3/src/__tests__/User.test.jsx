import React from 'react'
import {shallow} from 'enzyme'
import User from '../User'

describe('<User/>', () => {
  it('should return user with name Maks', () => {
    const user = {
        name: "Maks",
    }
    const wrappedComponent = shallow(<User name={user.name} />);

    expect(wrappedComponent.find('.user__name').exists()).toEqual(true);
  });
  
  it('should return user with age 25', () => {
    const user = {
        age: 25,
    }
    const wrappedComponent = shallow(<User age={user.age} />);

    expect(wrappedComponent.find('.user__age').exists()).toEqual(true);
  });
});
