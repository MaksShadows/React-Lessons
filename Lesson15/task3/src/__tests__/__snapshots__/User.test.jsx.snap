// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User/> should return user with age 25 1`] = `
<li
  className="user"
>
  <span
    className="user__name"
  />
  <span
    className="user__age"
  >
    25
  </span>
</li>
`;

exports[`<User/> should return user with name Maks 1`] = `
<li
  className="user"
>
  <span
    className="user__name"
  >
    Maks
  </span>
  <span
    className="user__age"
  />
</li>
`;
